### PR TITLE
fix kafka CVE-2023-25194, update kafka client to 3.4.0

### PR DESCRIPTION
**Pull request by xie-shujian**
### Description
[Describe what this change achieves]
* Category (Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation)
* Why these changes are required?
* What is the old behavior before changes and new behavior after changes?

### Issues Resolved
[List any issues this PR will resolve]
fix kafka CVE-2023-25194, update kafka client to 3.4.0
Is this a backport? If so, please add backport PR # and/or commits #
upgrade kafka client and spring test framework 
### Testing
[Please provide details of testing done: unit testing, integration testing and manual testing]
Test pass

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by scrawfor99** [comment]
Hi @xie-shujian, thank you for taking the time to open up a PR to update the Kafka dependency. Currently, we are having to wait on updating because there is not a compatible version of Springframework. You can see this on the original PR for the issue #2434.

File: .github/workflows/integration-tests.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        jdk: [11, 17]
+        jdk: [17]
```
Comment (peternied) [inline .github/workflows/integration-tests.yml:14]: For this bulk test job, why not run these tests on both platforms?

File: build.gradle [hunk 2]
```diff
@@ -24,7 +24,8 @@ buildscript {
         opensearch_build = version_tokens[0] + '.0'
 
         common_utils_version = System.getProperty("common_utils.version", '3.0.0.0-SNAPSHOT')
-        kafka_version  = '3.0.2'
+        kafka_version  = '3.4.0'
+        kafkaVersion   = '3.4.0'
```
Comment (peternied) [inline build.gradle:28]: Unused, lets remove

**Review by peternied — CHANGES_REQUESTED** [review]
Thank you for creating this pull request.

The DCO check failed, learn more about it and how to remedy from the failure https://github.com/opensearch-project/security/pull/2481/checks?check_run_id=11567975502

Also sounds like after this is resolved there will be the problem that @scrawfor99 mentioned, did you have thoughts on if this can be worked around?

**Comment by scrawfor99** [comment]
Hi @xie-shujian, as @peternied, I am happy to help out with this issue if you have any questions. That being said, it looks like I was mistaken about the version we need not being out yet. The website I looked at was outdated... 

If you are able to address the comments Peter left, we can get this squared away together. You will want to follow the guidance on this [question](https://github.com/spring-projects/spring-kafka/issues/2574).

File: .github/workflows/integration-tests.yml (hunk 1 above)
Comment (xie-shujian) [reply .github/workflows/integration-tests.yml:14]: I upgrade the spring test framework, it not support jdk 11, when I do the the test, it failed, raise an error a class version error. So I remove the jdk11, left jdk17
org.springframework.kafka:spring-kafka-test:3.0.3

**Comment by xie-shujian** [comment]
> Hi @xie-shujian, thank you for taking the time to open up a PR to update the Kafka dependency. Edit: Apparently there is a workaround to fix the version mismatch.

I know you work on it, and block by the spring test framework. I do more work to fix it.

**Comment by xie-shujian** [comment]
> Hi @xie-shujian, as @peternied, I am happy to help out with this issue if you have any questions. That being said, it looks like I was mistaken about the version we need not being out yet. The website I looked at was outdated...
> 
> If you are able to address the comments Peter left, we can get this squared away together. You will want to follow the guidance on this [question](https://github.com/spring-projects/spring-kafka/issues/2574).

I see this issue and flow the spring guide to solve it. spring test frame work can support kafka 3.0.4
there is an ext property, which I add 
kafkaVersion = '3.4.0' 
and one more package
testImplementation "org.apache.kafka:kafka-group-coordinator:${kafka_version}"

File: build.gradle (hunk 2 above)
Comment (xie-shujian) [reply build.gradle:28]: kafka_version  = '3.4.0' //for kafka client 
 kafkaVersion   = '3.4.0' // hacked spring kafka test framework or else it only support 3.3.2

**Comment by xie-shujian** [comment]
@scrawfor99 @peternied thanks for all your suggestion, I update something
now Only the [build.gradle](https://github.com/xie-shujian/security/blob/main/build.gradle) is update
test passed both jdk 11 and jkd 17
update kafka version to 3.4.0
update spring kafak test framework to 2.9.6
add one more coordinator jar
kafka_version  = '3.4.0' 
testImplementation "org.apache.kafka:kafka-group-coordinator:${kafka_version}"
testImplementation 'org.springframework.kafka:spring-kafka-test:2.9.6'

**Pull request closed without merging**